pkgs/test/lib/src/runner/browser/platform.dart

```diff
@@ -364,36 +432,23 @@ class BrowserPlatform extends PlatformPlugin
         }
       ''';
 
-      await _compilers.compile(bootstrapContent, jsPath, suiteConfig);
-      if (_closed) return;
-
-      var bootstrapUrl = p.toUri(p.relative(dartPath, from: _root)).path +
-          '.browser_test.dart';
-      _jsHandler.add(bootstrapUrl, (request) {
-        return shelf.Response.ok(bootstrapContent,
-            headers: {'Content-Type': 'application/dart'});
-      });
-
-      var jsUrl = p.toUri(p.relative(dartPath, from: _root)).path +
-          '.browser_test.dart.js';
-      _jsHandler.add(jsUrl, (request) {
-        return shelf.Response.ok(File(jsPath).readAsStringSync(),
-            headers: {'Content-Type': 'application/javascript'});
-      });
-
-      var mapUrl = p.toUri(p.relative(dartPath, from: _root)).path +
-          '.browser_test.dart.js.map';
-      _jsHandler.add(mapUrl, (request) {
-        return shelf.Response.ok(File(jsPath + '.map').readAsStringSync(),
-            headers: {'Content-Type': 'application/json'});
-      });
-
-      if (suiteConfig.jsTrace) return;
-      var mapPath = jsPath + '.map';
-      _mappers[dartPath] = JSStackTraceMapper(File(mapPath).readAsStringSync(),
-          mapUrl: p.toUri(mapPath),
-          sdkRoot: Uri.parse('org-dartlang-sdk:///sdk'),
-          packageMap: (await currentPackageConfig).toPackageMap());
+      await _compile(
+          bootstrapCode, dir, dartPath, '.browser_test.dart', suiteConfig);
+
+      if (workers.isNotEmpty) {
+        final dartDir = p.dirname(dartPath);
+
+        for (var worker in workers) {
```

Comment:
Agreed, with my proposal they'll be compiled each time they're included. I'd be happy with a `@CompileToJs()` too. The annotation could also be usefull in other use cases, e.g. right now I don't know how to have Flutter compile my workers to JS (I have to script that part). But the scope of the change goes way beyond `test`.

Reply:
The purpose of this contribution was to support test cases that involve Web Workers. It is easy to test Isolates on native platforms because Isolate entrypoints are top-level functions or static methods, plain Dart. For the Web platform, the story is slightly different and Web Worker entry points are JavaScript scripts. In Dart, they must be implemented as a `main()` function, and the file containing the `main()` function must be compiled to JavaScript so it can be loaded and executed by the browser.

In terms of design, I have proposed to support automatic compilation of Dart Web Worker files to JavaScript by leveraging the way `test` already compiles Dart test files to JavaScript automatically. My solution also relies on `test`'s HTML file template to reference the Dart Web Workers so `test` knows what it has to compile. So if a developer adds a tag like `<link rel="x-web-worker" href="path/to/web_worker.dart.js">` to the test template file, then `test` must compile `path/to/web_worker.dart` before running tests.

I have already implemented a basic test harness on top of `test` to support this use case (see https://github.com/d-markey/squadron/blob/main/test/squadron_js_test.dart#L43), but right now, JavaScript compilation must be run manually before executing tests. It's easy to forget that step, so I wanted to have `test` support it automatically and ended up with that proposition.

@jakemac53 righteously commented that Web Worker entry points could instead be marked with annotations. This would eliminate the need to customize the template file. And I believe this is a better proposition, because annotations are part of the source code and could be used for more use-cases than just testing.

For instance, when releasing a Flutter app that contains Web Workers, the Web Worker Dart files must also be compiled to JavaScript and deployed to the `web` folder. Currently, this part must be scripted to have Web Workers compiled in addition to the Flutter build process. I suppose it could also be useful for `flutter_test`.

With annotations to mark additional entry points, both `test` and Flutter could use them to have these assets compiled automatically :-)

So this contribution can be dropped and instead, I would love to have a more general mechanism to better support Web Workers in tests + release scenarios!